modules/catalog/src/main/java/org/apache/ignite/internal/catalog/commands/CatalogPrimaryKey.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.catalog.commands;
+
+import static org.apache.ignite.internal.lang.IgniteStringFormatter.format;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import org.apache.ignite.internal.catalog.CatalogValidationException;
+
+/** Base class of for a primary key. */
+public abstract class CatalogPrimaryKey {
```

Comment:
It can't be plural, since it _a_ primary key - and there are several flavours of it hash-index-backed / sorted-index-backed. At the moment only a table can have primary key. 
I think when something new comes up, that requires an its own primary it, then this entity can be renamed.

Reply:
Well, I think it still better to rename it. Renamed to `Table*PrimaryKey`.